src/ert/libres_facade.py

```diff
@@ -196,30 +194,18 @@ def get_data_key_for_obs_key(self, observation_key: str) -> str:
         else:
             return obs.data_key
 
-    def load_gen_data(
-        self,
-        ensemble: EnsembleReader,
-        key: str,
-        report_step: int,
-        realization_index: Optional[int] = None,
-    ) -> DataFrame:
-        realizations = ensemble.realization_list(RealizationStorageState.HAS_DATA)
-        if realization_index is not None:
-            if realization_index not in realizations:
-                raise IndexError(f"No such realization {realization_index}")
-            realizations = [realization_index]
-        try:
-            vals = ensemble.load_responses(key, tuple(realizations)).sel(
-                report_step=report_step, drop=True
-            )
-        except KeyError as e:
-            raise KeyError(f"Missing response: {key}") from e
-        index = pd.Index(vals.index.values, name="axis")
-        return pd.DataFrame(
-            data=vals["values"].values.reshape(len(vals.realization), -1).T,
-            index=index,
-            columns=realizations,
-        )
+    def get_gen_data_keys(self) -> List[str]:
+        ensemble_config = self.config.ensemble_config
+        gen_data_keys = ensemble_config.get_keylist_gen_data()
+        gen_data_list = []
+        for key in gen_data_keys:
```

Comment:
Think we would like to get away from this pattern, as this assumes that storage and the configuration are in sync, we should rather load the response keys from the experiment.

Reply:
This will be fixed when we refactor dark_storage.records